### ISSUE TITLE
K.O. flash effect not working properly with projectiles.

### ISSUE THREAD

**Issue by UrsoManso** [ikemen-engine/Ikemen-GO#339]
https://user-images.githubusercontent.com/77592586/130430097-05c06cf6-78b0-4269-84be-1fdf71604c84.mp4
Mugen 1.1

https://user-images.githubusercontent.com/77592586/130430131-c55f1d23-fa6f-454a-a578-c156e3476064.mp4
Ikemen

This issue also happens with projectiles from other characters but only when they return to state 0 before it hits, with Iori's projectile it doesn't work at all.

[Iori.zip](https://github.com/Windblade-GR01/Ikemen-GO/files/7031052/Iori.zip)

**Comment by K4thos** [comment]
problematic code, IORI.vns, statedef -2:
```ini
[State -2, Helper]
type = Helper
triggerall = !numhelper(50000) && !numhelper(50001)
triggerall = WinKO && var(23)
trigger1 = projhittime(0) = 1
trigger2 = movehit && stateno = [1300,1599]
trigger3 = movehit && stateno = 3602
trigger4 = movehit && stateno = 3402
trigger5 = numtarget(3400)
helpertype = normal
name = "K.O."
ID = 50001
stateno = 50000
pos = 160,120
postype = left    ;p2,front,back,left,right
facing = 1
keyctrl = 0
ownpal = 0
supermovetime = 999999999999
pausemovetime = 999999999999
ignorehitpause = 1
```

by the time WinKO returned 1 projhittime(0) already was at 2 instead of expected 1 (WinKO being detectable 1 frame to late compared to mugen).

This problem existed in all Ikemen GO and Ikemen SSZ versions.

Fixed by making WinKO detectable without delay (above linked commit). I've done decent amount of testing, so hopefully this won't lead to regressions.

**Comment by K4thos** [comment]
Unfortunately my attempt to fix this problem resulted in different issue - 1 frame difference between setting roundstate = 3 and updating matchover. Previous commit has been reverted, ticket should remain open.

**Comment by K4thos** [comment]
fixed by #572
